src/zenml/post_execution/pipeline.py

```diff
@@ -74,56 +70,51 @@ def get_pipeline(
 
     Args:
         pipeline: Name, class or instance of the pipeline.
-        version: Optional version of the pipeline. If not given, the latest
-            version will be returned.
-        **kwargs: The deprecated `pipeline_name` is caught as a kwarg to
-            specify the pipeline instead of using the `pipeline` argument.
+        version: Optional pipeline version. Behavior depends on `pipeline`:
+            - If `pipeline` is a pipeline instance, this argument is ignored.
+            - If `pipeline` is a pipeline class or name, then this argument
+            specifies the version of the pipeline to return. If not given, a
+            `PipelineClassView` will be returned that contains all versions and
+            all runs of this pipeline name/class.
 
     Returns:
-        A post-execution pipeline view for the given pipeline or `None` if
-        it doesn't exist.
+        A post-execution view for the given pipeline name, class, or instance
+        or `None` if it doesn't exist.
 
     Raises:
         RuntimeError: If the pipeline was not specified correctly.
     """
     from zenml.pipelines.base_pipeline import BasePipeline
 
-    if isinstance(pipeline, str):
-        pipeline_name = pipeline
-    elif isinstance(pipeline, BasePipeline):
+    # Pipeline instance: find the corresponding pipeline version in the DB.
+    if isinstance(pipeline, BasePipeline):
         pipeline_model = pipeline._get_registered_model()
         if pipeline_model:
             return PipelineView(model=pipeline_model)
         else:
             return None
+
+    # Otherwise, for pipeline name or class, determine the name first.
+    if isinstance(pipeline, str):
```

Comment:
```suggestion
    elif isinstance(pipeline, str):
```

Maybe, just to preserve the logic of this conditional, even if the `return` would bypass the rest?

Reply:
I think `if` is preferable over `elif` here since certain linters like pylint even complain if you do `elif` since the "else" part is effectively redundant